src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,43 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+
+  constructor(name, health = 100) {
+    this.name = name;
+    this.health = health;
+    Animal.alive.push(this);
+  }
+
+  checkHealth() {
+    if (this.health <= 0) {
+      Animal.alive = Animal.alive.filter((animal) => animal !== this);
+    }
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name, health = 100) {
+    super(name, health);
+    this.hidden = false;
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  constructor(name, health = 100) {
+    super(name, health);
+  }
+
+  bite(prey) {
+    if (prey instanceof Herbivore && !prey.hidden && prey.health > 0) {
+      prey.health -= 50;
+      prey.checkHealth();
+    }
```

Comment:
[CODE PERFOMANCE] - It's preferred to filter animals with health less than or equal to 0 from `Animal.alive` array using Array's `filter` method after the prey's health is decreased. The current implementation could cause performance issues due to the potential for `checkHealth` to be called multiple times in a loop. A more efficient approach would be to decrease the prey's health and then filter the `Animal.alive` array in one go, outside of the `bite` method.